2022/opmodes/JavaTeleop.java

```diff
@@ -0,0 +1,159 @@
+
+import team25core.DeadmanMotorTask;
+import team25core.GamepadTask;
+import team25core.MechanumGearedDrivetrain;
+import team25core.OneWheelDirectDrivetrain;
+import team25core.Robot;
+import team25core.RobotEvent;
+import team25core.RunToEncoderValueTask;
+import team25core.SingleShotTimerTask;
+import team25core.StandardFourMotorRobot;
+import team25core.TankMechanumControlScheme;
+import team25core.TankMechanumControlSchemeReverse;
+import team25core.TeleopDriveTask;
+
+@TeleOp(name = "JavaTeleop")
+//@Disabled
+public class JavaTeleop extends StandardFourMotorRobot {
+
+    private Telemetry.Item buttonTlm;
+
+    private TeleopDriveTask drivetask;
+    private DcMotor carouselMech;
+    private DcMotor liftMotor;
+    private DcMotor intakeMotor;
+
+    private DeadmanMotorTask liftMotorUpTask;
+    private DeadmanMotorTask liftMotorDownTask;
+
+    private DeadmanMotorTask intakeTask;
+    private DeadmanMotorTask outtakeTask;
+
+    private MechanumGearedDrivetrain drivetrain;
+
+    private static final int TICKS_PER_INCH = 79;
+
+    @Override
+    public void handleEvent(RobotEvent e) {
+    }
+
+    @Override
+    public void init() {
+
+        super.init();
+
+        //mechanisms
+        carouselMech = hardwareMap.get(DcMotor.class, "carouselMech");
+        liftMotor = hardwareMap.get(DcMotor.class,"liftMotor");
+        intakeMotor = hardwareMap.get(DcMotor.class, "intakeMotor");
+
+        // using encoders to record ticks
+
+        backLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        backRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontLeft.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        frontRight.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        //telemetry
+        buttonTlm = telemetry.addData("buttonState", "unknown");
+
+        TankMechanumControlSchemeReverse scheme = new TankMechanumControlSchemeReverse(gamepad1);
+        drivetrain = new MechanumGearedDrivetrain(motorMap);
+        drivetrain.setCanonicalMotorDirection();
+        // Note we are swapping the rights and lefts in the arguments below
+        // since the gamesticks were switched for some reason and we need to do
+        // more investigation
+        drivetask = new TeleopDriveTask(this, scheme, backLeft, backRight, frontLeft, frontRight);
+
+        liftMotorUpTask = new DeadmanMotorTask(this, liftMotor,  -0.5, GamepadTask.GamepadNumber.GAMEPAD_2, DeadmanMotorTask.DeadmanButton.LEFT_STICK_UP);
+        liftMotorDownTask    = new DeadmanMotorTask(this, liftMotor, 0.5, GamepadTask.GamepadNumber.GAMEPAD_2, DeadmanMotorTask.DeadmanButton.LEFT_STICK_DOWN);
+        intakeTask = new DeadmanMotorTask(this, intakeMotor,  -0.5, GamepadTask.GamepadNumber.GAMEPAD_2, DeadmanMotorTask.DeadmanButton.RIGHT_STICK_UP);
+        outtakeTask    = new DeadmanMotorTask(this, intakeMotor, 0.5, GamepadTask.GamepadNumber.GAMEPAD_2, DeadmanMotorTask.DeadmanButton.RIGHT_STICK_DOWN);
```

Comment:
Very nice.  Well done @shreyas61 

I will approve this and merge as soon as the style problems are fixed over in the terkel repository.